src/python/pants/init/logging.py

```diff
@@ -27,28 +27,32 @@ def init_rust_logger(log_level: LogLevel, log_show_rust_3rdparty: bool) -> None:
     native.init_rust_logging(log_level.level, log_show_rust_3rdparty)
 
 
-def setup_logging_to_stderr(python_logger: Logger, log_level: LogLevel) -> None:
-    """We setup logging as loose as possible from the Python side, and let Rust do the filtering."""
-    native = Native()
-    handler = NativeHandler.create(log_level, native, stream=sys.stderr)
-    python_logger.addHandler(handler)
-    # Let the rust side filter levels; try to have the python side send everything to the rust logger.
-    LogLevel.TRACE.set_level_for(python_logger)
-
-
 class NativeHandler(StreamHandler):
+    """This class is installed as a Python logging module handler (using  the logging.addHandler
+    method) and proxies logs to the Rust logging infrastructure."""
+
     def __init__(
         self,
         log_level: LogLevel,
-        native: Native,
         stream: Optional[TextIO] = None,
         native_filename: Optional[str] = None,
     ):
+
+        if stream is not None and native_filename is not None:
+            raise RuntimeError("NativeHandler must output to either a stream or a file, not both")
+
         super().__init__(stream)
-        self.native = native
+        self.native = Native()
```

Comment:
Again, safe because this is a singleton, correct? Possibly add a comment: `NB: Native() is a singleton.`

Reply:
We use `Native()` a lot and I think it would clutter up the code too much to point out in a comment everywhere it's used that it's a singleton. The class definition of `Native` in `native.py` makes it clear that it's a singleton.